### PR TITLE
KOGITO-778: [DMN Designer] Client-side marshalling tests

### PR DESCRIPTION
**Pull request by manstis**
@jomarko WDYT?

I've tweaked your PR to include a basic test of "round tripping" a diagram..

It'd be good to be the basis of more integration tests.

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/src/test/java/org/kie/workbench/common/dmn/showcase/client/selenium/DMNDesignerKogitoSeleniumIT.java [hunk 1]
```diff
@@ -86,12 +90,31 @@ public void teardown() {
...
     }
 
+    @Test
+    public void testNewDiagram() throws Exception {
+        setContent("");
+
+        final String xml = getContent();
+        assertThat(xml).isNotBlank();
+
+        System.out.println(xml);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/src/test/java/org/kie/workbench/common/dmn/showcase/client/selenium/DMNDesignerKogitoSeleniumIT.java]: I prefer compare with some specific text than just printing and watching jenkins logs :)

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml [hunk 2]
```diff
@@ -442,6 +457,16 @@
...
+    <dependency>
+      <groupId>org.xmlunit</groupId>
+      <artifactId>xmlunit-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.xmlunit</groupId>
+      <artifactId>xmlunit-assertj</artifactId>
+      <scope>test</scope>
+    </dependency>
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml:454]: Could you doublecheck please these are not in droolsjbpm-build-bootstrap?

**Comment by manstis** [comment]
@jomarko As an example I've added another test.

I changed the file extensions to be XML (simply so I can format then in IDEA).

I also found, when the test files contain line breaks, that there were errors in the WebDriver relating to unescaped line-breaks. Since _how_ the XML is sent to the WebDriver is a test implementation detail I simply replaced the join with line-breaks to join (as one long String).

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/src/test/java/org/kie/workbench/common/dmn/showcase/client/selenium/DMNDesignerKogitoSeleniumIT.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/src/test/java/org/kie/workbench/common/dmn/showcase/client/selenium/DMNDesignerKogitoSeleniumIT.java]: Oh sure.. this test would need refactoring to check the returned XML matches a "new diagram" template.. let me do that now (in a new commit to follow).

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml (hunk 2 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml:454]: Will do.. this is PoC (as these would need moving to `droolsjbpm-build-bootstrap` too).

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml:454]: Yes `org.xmlunit:xmlunit-core` is in `kie-parent` but version `2.3.0`. I'll submit a PR to you for `droolsjbpm-build-bootstrap` pushing to 2.6.3 (hopefully it won't break things). `org.xmlunit:xmlunit-assertj` is not declared yet.

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml:454]: I have a PR ready to submit to `jomarko/droolsjbpm-build-bootstrap` but it's not listed in GitHub.

IDK if you have to grant me access? I could create this one for `jomarko/kie-wb-common` OK.

https://github.com/manstis/droolsjbpm-build-bootstrap/tree/KOGITO-778

**Comment by manstis** [comment]
@jomarko Sorry to hijack your PR to this degree..

I've added tests for round-tripping all the DMN node types and squashed into a single commit. WDYT?

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml (hunk 2 above)
Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml:454]: not sure, created it here https://github.com/jomarko/droolsjbpm-build-bootstrap/pull/1/files

**Comment by jomarko** [comment]
@manstis could you resolve conflicts?

**Comment by manstis** [comment]
@jomarko Seems OK now; thanks.

**Comment by jomarko** [comment]
@manstis why the added resources are xml and not dmn extension?

**Comment by manstis** [comment]
@jomarko See my comment [above](https://github.com/jomarko/kie-wb-common/pull/1#issuecomment-578141591) (about why DMN files have the `.xml` extension).

I have no strong opinion about renaming back to `.dmn`.

**Pull request merged**